exa-mcp-server/src/main.ts

```
@@ -0,0 +1,70 @@
+/**
+ * MCP Server - Main Entry Point
+ *
+ * This file serves as the entry point for the MCP Server Actor.
+ * It sets up a proxy server that forwards requests to the locally running
+ * MCP server, which provides a Model Context Protocol (MCP) interface.
+ */
+
+// Apify SDK - toolkit for building Apify Actors (Read more at https://docs.apify.com/sdk/js/)
+import { Actor, log } from 'apify';
+
+import { startServer } from './server.js';
+
+// This is an ESM project, and as such, it requires you to specify extensions in your relative imports
+// Read more about this here: https://nodejs.org/docs/latest-v18.x/api/esm.html#mandatory-file-extensions
+// Note that we need to use `.js` even when inside TS files
+// import { router } from './routes.js';
+
+// Configuration constants for the MCP server
+// Use a remote Exa MCP server via mcp-remote with streamable (http) transport.
+// EXA_API_KEY is optional; if provided, it's appended as exaApiKey query param.
+const { EXA_API_KEY } = process.env;
+const EXA_URL_BASE = 'https://mcp.exa.ai/mcp';
+const EXA_URL = EXA_API_KEY ? `${EXA_URL_BASE}?exaApiKey=${encodeURIComponent(EXA_API_KEY)}` : EXA_URL_BASE;
```

It is not going to work without EXA_API_KEY, right? So I would rather fail early than later.